src/assembler/encoder.ts

```diff
@@ -279,12 +255,16 @@ function writeDataInstruction(
   if (isSymbolDataInstruction(instruction)) {
     const bytes = Word.fromUnsignedInteger(0x0).toBytes()
     writer.align(4)
+    writer.mapLine(instruction.line)
+    writer.addLabel(instruction.label, instruction.line)
     for (const option of instruction.options) {
       writer.addDataRelocation(option.trim(), bytes.length, instruction.line)
       writer.writeBytes(bytes)
     }
     return
   } else if (instruction.name === 'ALIGN') {
+    writer.mapLine(instruction.line)
+    writer.addLabel(instruction.label, instruction.line)
```

Comment:
can ALIGN have a label?

Reply:
additionally everywhere else you aligned first and then called mapLine and addLabel. Is this intentional?

Reply:
You are right. It does not make sense to add a label or line mapping for a `ALIGN` instruction. I therefor removed it from the code.